gtk4/src/dialog.rs

```diff
@@ -44,17 +45,41 @@ pub trait DialogExtManual: 'static {
     /// Shows the dialog and returns a `Future` that resolves to the
     /// `ResponseType` on response.
     ///
-    /// ```ignore
-    /// let dialog = gtk::MessageDialogBuilder::new()
-    ///    .buttons(gtk::ButtonsType::OkCancel)
+    /// ```no_run
+    /// use gtk4::prelude::*;
+    ///
+    /// # async fn run() {
+    /// let dialog = gtk4::MessageDialogBuilder::new()
+    ///    .buttons(gtk4::ButtonsType::OkCancel)
     ///    .text("What is your answer?")
     ///    .build();
     ///
     /// let answer = dialog.run_future().await;
     /// dialog.close();
     /// println!("Answer: {:?}", answer);
+    /// # }
     /// ```
     fn run_future<'a>(&'a self) -> Pin<Box<dyn Future<Output = ResponseType> + 'a>>;
+
+    // rustdoc-stripper-ignore-next
+    /// Shows the dialog and calls the callback when a response has been received.
+    ///
+    /// **Important**: this function isn't blocking.
+    ///
+    /// ```no_run
+    /// use gtk4::prelude::*;
+    ///
+    /// let dialog = gtk4::MessageDialogBuilder::new()
+    ///    .buttons(gtk4::ButtonsType::OkCancel)
+    ///    .text("What is your answer?")
+    ///    .build();
+    ///
+    /// dialog.run(|obj, answer| {
+    ///     obj.close();
+    ///     println!("Answer: {:?}", answer);
+    /// });
+    /// ```
+    fn run<F: FnOnce(&Self, ResponseType) + 'static>(&self, f: F);
```

Comment:
Should probably be called `run_async()` because it's not like the original `run()`.

Reply:
ebassi suggested response as a name, which sounds better to me instead of calling it "run"

Reply:
"response" won't work https://gtk-rs.org/gtk4-rs/gtk4/trait.DialogExt.html#tymethod.response

Reply:
Ah :/ I wonder why some signals are named "$signal_name" like the response here while others are named "emit_$signal_name" like the `emit_close`.

Reply:
`emit_XXX()` are the ones we autogenerated if a signal is marked as action signal. `response()` has a emission function in C that is named like that.